assets/src/edit-story/app/history/useHistoryReducer.js

```diff
@@ -42,9 +42,35 @@ const reducer = (size) => (state, { type, payload }) => {
         );
 
         if (isReplay) {
+          const currentEntry = state.entries[state.offset];
+          const offset = state.entries.indexOf(state.replayState);
+          // If a page has changed and it was not an added page, and if the page is about to change with replay,
+          // Ensure the user stays on the page where the latest change happened.
+          if (
+            currentEntry.pages !== state.replayState.pages &&
+            currentEntry.pages.length === state.replayState.pages.length &&
+            currentEntry.current !== state.replayState.current
+          ) {
+            const changedPage = currentEntry.pages.filter((page, index) => {
+              return page !== state.replayState.pages[index];
+            });
+            // If a changed page was found.
+            if (changedPage.length === 1) {
+              // Ensure we stay on the changed page by overriding the previously saved current page.
+              const current = changedPage[0].id;
+              const replayState = { ...state.replayState, current };
+              const entries = state.entries;
+              Object.assign([], entries, { [offset]: replayState });
```

Comment:
What's the meaning of this line?

Reply:
Oh!
Unless that was a philosophical question then this line seems to have no meaning and should be removed.

Reply:
That's what I though. Unless you actually wanted to merge new `[offset]` value. Then it'd have to be rewritten.